### ISSUE TITLE
terraform destroy doesn't work properly for Amazon EKS cluster

### ISSUE THREAD

**Issue by ruzickap** [rancher/terraform-provider-rancher2#858]
Hello.

I'm have a simple terraform code which should create + delete Amazon EKS:

```hcl
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 3.73.0"
    }
    rancher2 = {
      source  = "rancher/rancher2"
      version = "1.22.2"
    }
  }
}

provider "rancher2" {
  api_url   = "https://rancher.xxx.k8s.xxxx.dev"
  token_key = "token-xxxxx:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
}

provider "aws" {
  region = "eu-west-2"
}

resource "aws_iam_user" "rancher" {
  name = "rancher-test-eks"
}

resource "aws_iam_user_policy_attachment" "rancher" {
  user       = aws_iam_user.rancher.name
  policy_arn = "arn:aws:iam::aws:policy/AdministratorAccess"
}

resource "aws_iam_access_key" "rancher" {
  user = aws_iam_user.rancher.name
}

resource "rancher2_cloud_credential" "rancher_aws_account" {
  name        = "aws-rancher-test-eks"
  description = "Access to AWS account for test-eks"
  amazonec2_credential_config {
    access_key = aws_iam_access_key.rancher.id
    secret_key = aws_iam_access_key.rancher.secret
  }
}

resource "rancher2_cluster" "cluster" {
  name        = "test-eks"

  eks_config_v2 {
    cloud_credential_id = rancher2_cloud_credential.rancher_aws_account.id
    kubernetes_version  = "1.21"
    name                = "test-eks"
    private_access      = true
    public_access       = true
    region             = "eu-west-2"

    node_groups {
      instance_type = "t2.medium"
      desired_size = 2
      max_size = 2
      min_size = 2
      name     = "test-eks-ng"
    }
  }
}
```

When I run `terraform apply` the EKS cluster is properly created + registered to Rancher.
`terrafrom destroy` doesn't give me any error, but I can see the cluster in AWS console:

```text
Plan: 0 to add, 0 to change, 5 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

rancher2_cluster.cluster: Destroying... [id=c-5czcb]
aws_iam_user_policy_attachment.rancher: Destroying... [id=rancher-test-eks-20220201151916128900000001]
aws_iam_user_policy_attachment.rancher: Destruction complete after 1s
rancher2_cluster.cluster: Destruction complete after 4s
rancher2_cloud_credential.rancher_aws_account: Destroying... [id=cattle-global-data:cc-pwj5b]
rancher2_cloud_credential.rancher_aws_account: Destruction complete after 1s
aws_iam_access_key.rancher: Destroying... [id=AKIxxxxxxxxxxxxxxxx5O]
aws_iam_access_key.rancher: Destruction complete after 0s
aws_iam_user.rancher: Destroying... [id=rancher-test-eks]
aws_iam_user.rancher: Destruction complete after 2s

Destroy complete! Resources: 5 destroyed.
```

The only part of the cluster which was really deleted was "EKS" Node Group.

It looks like terraform "destroy" doesn't seems to be working right...

**Comment by rawmind0** [comment]
Hi @ruzickap , the tf provider is just calling the Rancher API to create or remove the cluster. If the terraform destroy is not getting any error, may be the issue is at Rancher level. On our tests, the clusters where removed properly. Have you check your Rancher logs??

**Comment by ruzickap** [comment]
I'm not sure if the problem s somewhere in "terraform Rancher provider" or Rancher. Anyway I created https://github.com/rancher/rancher/issues/36353 to have it tracked in Rancher.

I have few more notes:
* I create the Amazon EKS cluster using the TF code above and everything is working fine
* If I then run `terrafrom destroy` I got the errors mentioned in: https://github.com/rancher/rancher/issues/36353
* If I use the Rancher Web interface (instead of running `terraform destroy`) to delete the EKS cluster - then the cluster is properly deleted with all the components.

Rancher error log produced by `terraform destroy`:

```text
$ stern -n cattle-fleet-local-system -n cattle-fleet-system -n cattle-system --tail 1 .
...
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Stopping cluster agent for c-7st6l
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=RoleBinding workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=Role workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=ResourceQuota workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=Namespace workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=ServiceAccount workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=LimitRange workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=ClusterRole workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=Secret workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down apiregistration.k8s.io/v1, Kind=APIService workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=Node workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=ConfigMap workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] Shutting down /v1, Kind=ConfigMap workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] [mgmt-auth-users-controller] Deleting clusterRoleTemplateBinding u-lb2u2o7x6j-admin for user u-lb2u2o7x6j
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-avoo7cnip5
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] [mgmt-auth-users-controller] Deleting globalRoleBinding grb-49gdg for user u-lb2u2o7x6j
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding u-lb2u2o7x6j-admin-cluster-owner in namespace p-ckdhf for crtb u-lb2u2o7x6j-admin
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding u-lb2u2o7x6j-admin-cluster-owner in namespace p-2rs4z for crtb u-lb2u2o7x6j-admin
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:31 [INFO] [mgmt-cluster-rbac-remove] Deleting namespace c-7st6l
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:29:32Z" level=info msg="deleting cluster [c-7st6l]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:29:32Z" level=info msg="starting node group deletion for config [ruzickap-eks]"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [ERROR] error syncing 'c-7st6l': handler mgmt-cluster-rbac-remove: clusters.management.cattle.io "c-7st6l" not found, requeuing
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [INFO] [mgmt-auth-users-controller] Deleting clusterRoleTemplateBinding c-7st6l-fleet-default-owner for user u-ydnafvbzey
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [INFO] [mgmt-auth-users-controller] Deleting token u-ydnafvbzey for user u-ydnafvbzey
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-laonr7f4r3
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-7st6l-fleet-default-owner-cluster-owner in namespace p-ckdhf for crtb c-7st6l-fleet-default-owner
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-7st6l-fleet-default-owner-cluster-owner in namespace p-2rs4z for crtb c-7st6l-fleet-default-owner
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:32 [ERROR] error syncing 'fleet-default/c-7st6l': handler cluster-create: failed to delete c-7st6l/c-7st6l-fleet-default-owner management.cattle.io/v3, Kind=ClusterRoleTemplateBinding for cluster-create fleet-default/c-7st6l: clusterroletemplatebindings.management.cattle.io "c-7st6l-fleet-default-owner" not found, requeuing
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:37 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-f5nqk6o5bj
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:37 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding creator-cluster-owner-cluster-owner in namespace p-2rs4z for crtb creator-cluster-owner
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:37 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding creator-cluster-owner-cluster-owner in namespace p-ckdhf for crtb creator-cluster-owner
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:37 [INFO] [mgmt-project-rbac-remove] Deleting namespace p-ckdhf
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:37 [INFO] [mgmt-project-rbac-remove] Deleting namespace p-2rs4z
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:29:42Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:43 [INFO] [mgmt-auth-prtb-controller] Updating owner label for roleBinding crb-7dctyvj4qz
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 07:29:43 [INFO] [mgmt-auth-prtb-controller] Deleting roleBinding crb-7dctyvj4qz
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:29:53Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:30:03Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:30:13Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:30:23Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:30:33Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:30:44Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:30:54Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:04Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:15Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:25Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:35Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="waiting for config [c-7st6l] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=error msg="error syncing 'cattle-global-data/c-7st6l': handler eks-controller-remove: error deleting nodegroups for config [ruzickap-eks], requeuing"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="deleting cluster [c-7st6l]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="starting node group deletion for config [ruzickap-eks]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=error msg="error syncing 'cattle-global-data/c-7st6l': handler eks-controller-remove: error deleting nodegroups for config [ruzickap-eks], requeuing"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="deleting cluster [c-7st6l]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="starting node group deletion for config [ruzickap-eks]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=error msg="error syncing 'cattle-global-data/c-7st6l': handler eks-controller-remove: error deleting nodegroups for config [ruzickap-eks], requeuing"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="deleting cluster [c-7st6l]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=info msg="starting node group deletion for config [ruzickap-eks]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T07:31:45Z" level=error msg="error syncing 'cattle-global-data/c-7st6l': handler eks-controller-remove: error deleting nodegroups for config [ruzickap-eks], requeuing"
...
<last 3 lines `error deleting nodegroups for config` repeats all the time>
...
```

Logs produced by Rancher when deleting EKS cluster created by Terraform (code above) using Rancher Web GUI (this operation is successful):

```text
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Stopping cluster agent for c-8hn85
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=Secret workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=ClusterRole workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=Namespace workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=ServiceAccount workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down apiregistration.k8s.io/v1, Kind=APIService workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=LimitRange workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=Role workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down rbac.authorization.k8s.io/v1, Kind=RoleBinding workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=ResourceQuota workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=Node workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=ConfigMap workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] Shutting down /v1, Kind=ConfigMap workers
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] [mgmt-auth-users-controller] Deleting clusterRoleTemplateBinding u-edeu62rhnx-admin for user u-edeu62rhnx
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-5uau4hrj2g
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] [mgmt-auth-users-controller] Deleting globalRoleBinding grb-l922n for user u-edeu62rhnx
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding u-edeu62rhnx-admin-cluster-owner in namespace p-vdv6m for crtb u-edeu62rhnx-admin
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:42 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding u-edeu62rhnx-admin-cluster-owner in namespace p-fxdtg for crtb u-edeu62rhnx-admin
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [INFO] [mgmt-cluster-rbac-remove] Deleting namespace c-8hn85
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:48:43Z" level=info msg="deleting cluster [c-8hn85]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:48:43Z" level=info msg="starting node group deletion for config [ruzickap-eks]"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [ERROR] error syncing 'c-8hn85': handler mgmt-cluster-rbac-remove: clusters.management.cattle.io "c-8hn85" not found, requeuing
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [INFO] [mgmt-auth-users-controller] Deleting clusterRoleTemplateBinding c-8hn85-fleet-default-owner for user u-ynr3kypexf
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-r535l4fewb
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [INFO] [mgmt-auth-users-controller] Deleting token u-ynr3kypexf for user u-ynr3kypexf
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-8hn85-fleet-default-owner-cluster-owner in namespace p-vdv6m for crtb c-8hn85-fleet-default-owner
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [INFO] [mgmt-auth-crtb-controller] Deleting rolebinding c-8hn85-fleet-default-owner-cluster-owner in namespace p-fxdtg for crtb c-8hn85-fleet-default-owner
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [ERROR] error syncing 'cluster/fleet-default/c-8hn85': handler auth-prov-v2-roletemplate: failed to update fleet-default/crt-c-8hn85-cluster-admin rbac.authorization.k8s.io/v1, Kind=Role for auth-prov-v2-roletemplate-c-8hn85 cluster-admin: roles.rbac.authorization.k8s.io "crt-c-8hn85-cluster-admin" not found, requeuing
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:43 [ERROR] error syncing 'fleet-default/c-8hn85': handler cluster-create: failed to delete fleet-default/c-8hn85-kubeconfig /v1, Kind=Secret for cluster-create fleet-default/c-8hn85: secrets "c-8hn85-kubeconfig" not found, requeuing
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:48 [INFO] [mgmt-project-rbac-remove] Deleting namespace p-fxdtg
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:48 [INFO] [mgmt-project-rbac-remove] Deleting namespace p-vdv6m
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:49 [INFO] [mgmt-auth-crtb-controller] Deleting roleBinding crb-qp63ht4pej
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:48:53Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:54 [INFO] [mgmt-auth-prtb-controller] Updating owner label for roleBinding crb-3o4ks27b42
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:54 [INFO] [mgmt-auth-prtb-controller] Deleting roleBinding crb-3o4ks27b42
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:56 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:48:56 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:49:04Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:49:09 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:49:09 [INFO] error in remotedialer server [400]: websocket: close 1006 (abnormal closure): unexpected EOF
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:49:14 [ERROR] Error during subscribe websocket: close sent
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:49:14 [ERROR] Error during subscribe websocket: close sent
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:49:14Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:49:24Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:49:34Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:49:45Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:49:55Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:50:05Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:50:15Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:50:26Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:50:36Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:50:46Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:50:56Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:51:06Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:51:13 [INFO] checking cluster [c-f7sj8] upstream state for changes
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:51:13 [INFO] updating cluster [c-f7sj8], upstream change detected
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:51:13 [INFO] change detected for cluster [c-f7sj8], updating spec
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:51:13 [INFO] change detected for cluster [c-f7sj8], updating EKSClusterConfig
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:51:17Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:51:27Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:51:37Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:51:47Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:51:58Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:52:08Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:52:18Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:52:28Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:52:39Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:52:49Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:52:59Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:53:09Z" level=info msg="waiting for config [c-8hn85] node groups to delete"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:53:09Z" level=info msg="deleting common launch template for config [c-8hn85]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:53:09Z" level=info msg="starting control plane deletion for config [c-8hn85]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:53:10Z" level=info msg="deleting service role for config [c-8hn85]"
cattle-system eks-config-operator-6bfb5548b7-8gtf4 eks-operator time="2022-02-02T08:53:10Z" level=info msg="deleting node instance role for config [c-8hn85]"
cattle-fleet-system fleet-controller-6666887949-l259x fleet-controller time="2022-02-02T08:55:33Z" level=info msg="Deleting expired registration secret cattle-fleet-clusters-system/c-f716d607c794f97d53a6bee06857b4357d0e9fc7974e1c9d0296e3a60a287"
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:56:13 [INFO] checking cluster [c-f7sj8] upstream state for changes
cattle-system rancher-6799f8f646-x4fs7 rancher 2022/02/02 08:56:13 [INFO] cluster [c-f7sj8] matches upstream, skipping spec sync
```

In short: 
* When deleting Amazon EKS cluster using `terraform destroy` - then it is failing
* When deleting Amazon EKS cluster using "Rancher GUI deleted" - this is working

**Comment by adam-sdk** [comment]
same issue !

**Comment by matttrach** [comment]
The last comment on this issue was 2 years ago, but I think this related to how Rancher returns a false positive on the object, then works to remove it in the background. Terraform moves on to destroying the authentication and without that authentication Rancher is unable to complete the destroy operation. One workaround would be to add a destroy time provisioner to the resource which runs a local_exec to sleep for a few minutes or even to poll Rancher for some confirmation that the cluster has actually be removed before destroying the credential. The original post is to older versions of Rancher and the provider, as such, I am going to close this issue as stale, but feel free to open a new issue if this is still a valid problem you are facing.
